crates/ide/src/hover/render.rs

```diff
@@ -381,6 +381,21 @@ where
     (label, docs)
 }
 
+fn const_label_value_and_docs(
+    db: &RootDatabase,
+    konst: Const,
+) -> (String, Option<hir::Documentation>) {
+    let label = if let Some(expr) = konst.value(db) {
+        format!("{} = {}", konst.display(db), expr)
+    } else {
+        konst.display(db).to_string()
+    };
+
+    let docs = konst.attrs(db).docs();
+
+    (label, docs)
+}
```

Comment:
So ideally we would be able to evaluate constants values, though that is currently not doable for us.

So what we should do here is special case const hovers to show the value *only* if its expressions is a literal, otherwise don't show the body at all as the hover could get pretty messy in other cases.

Reply:
I thought about this - but a lot of our constants are also things like Durations which would be useful

Reply:
We could also limit it to simple function calls, but we shouldn't render anything too complicating(and especially nothing that spans more than one line).

Reply:
Ye lets just go with how it is for now, maybe I'm looking at this wrong. Could you add/change a test that has a more complex const body though than just a literal or function call?

bors d+

Reply:
Can you give me an example, because honestly, grepping through consts in all of our code, we don't *practically* do anything more than literals and function calls. Did you have anything in mind?

Reply:
Just something like 
```rs
const FOO: usize = {
    let x = foo();
    x + 100
};
```
it doesn't have to compile(as in `foo` doesn't have to be a thing).

Reply:
Here's a really long const, and how it renders in vscode. Not *too* bad...

![image](https://user-images.githubusercontent.com/5489149/142272909-9eb4e867-bc1d-4fa7-af7a-9d01c5395c8d.png)

Reply:
Whew, I mean the important parts are at the top(the type of the const) so this should be fine 👍